### DISCUSSION TITLE
[Oracle] Behaviour filter on top differs from filter in navigation tree (per object type)

### DISCUSSION THREAD

**Discussion by F0xD8rk** [dbeaver/dbeaver#12178] [Q&A]
**Describe the solution you'd like**
If you set the result set size to f.e. 250 records, the filter on top does not show objects that you want to filter that are not in the first fetched 250 records of the object types. If you click on the "More" option under f.e. the object type _tables_, then the object with the filtered name is shown correctly.

![afbeelding](https://user-images.githubusercontent.com/79097587/109468876-89fcc480-7a6d-11eb-9340-cf872d2d6f52.png)

However, if you use the direct filter on an object type, the object is shown regardless of the result set size. Would it be possible to also show the object using the filter on top, regardless of the result set size?

![afbeelding](https://user-images.githubusercontent.com/79097587/109468764-5cb01680-7a6d-11eb-87fc-3bfed0c84d1e.png)

**Comment by uslss** [comment]
Unfortunately, it is not really possible. It could take too much time to load the whole tree in DB navigator.
You could use e.g. Tables folder filter instead
It could be found in connection properties
![image](https://user-images.githubusercontent.com/31996417/111489904-9fe7d600-874b-11eb-9c1e-934b3880f6b5.png)
or in the Connection context menu in DB navigator
![image](https://user-images.githubusercontent.com/31996417/111490103-c9086680-874b-11eb-84a3-21919289d6df.png)

**Comment by F0xD8rk** [comment]
This change would probably require you to redesign the DB Navigator, so I understand this would be difficult. 
At the moment I already use the suggested object filter to filter the results, so I will keep using this filter in the future.

The option to increase the default result set size using the preferences could also be a solution. However, this would probably cause performance to drop severely because of the increased result set size? (especially since we have +- 6000 tables in our database)

Thanks.
